### PR TITLE
Redesign Wallet navigation and merge assets and NFTs into single section

### PR DESCRIPTION
**Pull request by simoarpe**
### Light Mode 

Screenshot 2023-08-22 at 6 10 01 PM

### Dark Mode

Screenshot 2023-08-22 at 6 09 36 PM

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Open Brave Wallet
* Observe NFTs section is not present anymore in the bottom navigation view
* Observe the new tab layout is present at the top
* Tap on NFTs tab
* Observe the NFT grid is shown

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/AssetsFragment.java [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+import org.chromium.ui.widget.Toast;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Assets fragment displayed under portfolio section of Brave Wallet.
+ */
+public class AssetsFragment extends Fragment implements OnWalletListItemClick, ApprovedTxObserver {
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/AssetsFragment.java:71]: `AssetsFragment` class is the old `PortfolioFragment` renamed.

File: android/java/res/layout/fragment_portfolio.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (simoarpe) [inline android/java/res/layout/fragment_portfolio.xml:2]: Old portfolio layout has been moved to `fragment_assets.xml`

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java [hunk 3]
```diff
@@ -5,453 +5,57 @@
...
-
-    private void updatePendingTxNotification() {
-        Activity activity = getActivity();
-        if (activity instanceof BraveWalletActivity)
-            ((BraveWalletActivity) activity).showPendingTxNotification(hasPendingTx());
-    }
-
-    private NetworkModel getNetworkModel() {
-        return mWalletModel.getCryptoModel().getNetworkModel();
+        mViewPagerAdapter.getAssetsFragment().callAnotherApproveDialog();
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:59]: Do we need to move the pending transactions logic in a nested asset fragment? Better to keep it in the parent `Portfolio` fragment.

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/PortfolioFragmentStateAdapter.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    public CharSequence getPageTitle(final int position) {
+        return mTitles.get(position);
+    }
+
+    @NonNull
+    public AssetsFragment getAssetsFragment() {
+        if (mAssetsFragment == null) {
+            mAssetsFragment = AssetsFragment.newInstance();
+        }
+        return mAssetsFragment;
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/adapters/PortfolioFragmentStateAdapter.java:74]: This should always return the fragment in use than returning a new fragment. This indirect dependency can be avoided by keeping the pending tx logic in Portfolio/Parent fragment.

Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/adapters/PortfolioFragmentStateAdapter.java:74]: 👍

**Comment by simoarpe** [comment]
Feedback addressed @Pavneet-Sing

**Review by Pavneet-Sing — APPROVED** [review]

**Pull request merged**